Sources/Socket.swift

```diff
@@ -3128,191 +3123,131 @@ public class Socket: SocketReader, SocketWriter {
-	}
-	
-	///
-	/// Private method that reads all available data on an open socket into storage.
-	///
-	/// - Returns: number of bytes read.
-	///
-	private func readDatagramIntoStorage() throws -> (bytesRead: Int, fromAddress: Address?) {
-		
-		// Clear the buffer...
-		self.readBuffer.deinitialize()
-		self.readBuffer.initialize(to: 0x0)
-		memset(self.readBuffer, 0x0, self.readBufferSize)
-		var address: Address? = nil
-		
-		let addr = sockaddr_storage()
-		var length = socklen_t(MemoryLayout<sockaddr_storage>.size)
-		var addrPtr = addr.asAddr()
-		
-		// Read all the available data...
-		var count: Int = 0
-		repeat {
-			
-			#if os(Linux)
-				count = Glibc.recvfrom(self.socketfd, self.readBuffer, self.readBufferSize, 0, &addrPtr, &length)
-			#else
-				count = Darwin.recvfrom(self.socketfd, self.readBuffer, self.readBufferSize, 0, &addrPtr, &length)
-			#endif
-			
+
 			// Check for error...
 			if count < 0 {
-				
+
 				// - Could be an error, but if errno is EAGAIN or EWOULDBLOCK (if a non-blocking socket),
 				//		it means there was NO data to read...
 				if errno == EAGAIN || errno == EWOULDBLOCK {
-					
-					return (0, address)
+
+					return (0, nil)
 				}
-				
+
 				// - Handle a connection reset by peer (ECONNRESET) and throw a different exception...
 				if errno == ECONNRESET {
-					
+
 					throw Error(code: Socket.SOCKET_ERR_CONNECTION_RESET, reason: self.lastError())
 				}
-				
+
 				// - Something went wrong...
 				throw Error(code: Socket.SOCKET_ERR_RECV_FAILED, reason: self.lastError())
 			}
-			
-			if count == 0 {
-				
+
+			if count == 0 && sig.proto != .udp {
+
 				self.remoteConnectionClosed = true
 				return (0, nil)
 			}
-			
+
 			if count > 0 {
 				self.readStorage.append(self.readBuffer, length: count)
 			}
-			
-			// Retrieve the address...
-			if addrPtr.sa_family == sa_family_t(AF_INET6) {
-				
-				var addr = sockaddr_in6()
-				memcpy(&addr, &addrPtr, Int(MemoryLayout<sockaddr_in6>.size))
-				address = .ipv6(addr)
-				
-			} else if addrPtr.sa_family == sa_family_t(AF_INET) {
-				
-				var addr = sockaddr_in()
-				memcpy(&addr, &addrPtr, Int(MemoryLayout<sockaddr_in>.size))
-				address = .ipv4(addr)
-				
-			} else {
-				
-				throw Error(code: Socket.SOCKET_ERR_WRONG_PROTOCOL, reason: "Unable to determine receiving socket protocol family.")
-			}
-			
+
 			// Didn't fill the buffer so we've got everything available...
 			if count < self.readBufferSize {
-				
+
 				break
 			}
 
-			
-		} while count > 0
-		
+		} while count > 0 && sig.proto != .udp
+
+		// Retrieve the address...
+		if addrPtr.sa_family == sa_family_t(AF_INET6) {
+
+			var addr = sockaddr_in6()
+			memcpy(&addr, &addrPtr, Int(MemoryLayout<sockaddr_in6>.size))
+			address = .ipv6(addr)
+
+		} else if addrPtr.sa_family == sa_family_t(AF_INET) {
+
+			var addr = sockaddr_in()
+			memcpy(&addr, &addrPtr, Int(MemoryLayout<sockaddr_in>.size))
+			address = .ipv4(addr)
+
+		} else {
+
+			address = nil
+		}
+
```

Comment:
This whole block of code is **not needed* for TCP sockets (which will probably be used the most).  Why incur this processing on every TCP read?